### ISSUE TITLE
Album and artist missing

### ISSUE THREAD

**Issue by gburca** [v00d00/mediatomb#26]
On a new scan of the music files with a 11dea7337 build, all the artist names, album titles, and years show up as "Unknown". This happens when import.js is used.

My import.js tries to obtain the artist from `obj.meta[M_ARTIST]` with fall-back to `obj.aux["TPE2"]`. The `config.xml` file contains `<add-data tag="TPE2"/>` for both id3 and libextractor.

**Comment by v00d00** [comment]
Interesting, I'm travelling today so will take a look this evening.

**Comment by gburca** [comment]
The following properties are OK: title, upnpclass, dc:description, mimetype, location

The following properties are not OK: dc:title, upnp:artist, upnp:album, dc:date, etc...

Seems like there's something wrong with the metadata passed to the JS.

**Comment by gburca** [comment]
Found the problem. My taglib was too old (the default version in Ubuntu 16.04 is 1.9.1). Even though `WITH_TAGLIB` was set, cmake didn't set `HAVE_TAGLIB` because it couldn't find the right version. I wish it would have failed with a more obvious message. Instead it just issued a message that I didn't notice until later after I already debugged the problem.

This applies to all the `WITH_XYZ` options. It would help if cmake failed with: "You asked for XYZ but we couldn't find it. Either install XYZ or unset WITH_XYZ."
